orchestrator/oracle/oracle.go

```diff
@@ -0,0 +1,475 @@
+package oracle
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/rs/zerolog"
+	"golang.org/x/sync/errgroup"
+
+	ummedpforacle "github.com/umee-network/umee/price-feeder/oracle"
+	umeedpfprovider "github.com/umee-network/umee/price-feeder/oracle/provider"
+	umeedpftypes "github.com/umee-network/umee/price-feeder/oracle/types"
+	ummedpfsync "github.com/umee-network/umee/price-feeder/pkg/sync"
+)
+
+// We define tickerTimeout as the minimum timeout between each oracle loop.
+const (
+	tickerTimeout        = 1000 * time.Millisecond
+	availablePairsReload = 24 * time.Hour
+	BaseSymbolETH        = "ETH"
+)
+
+var (
+	// deviationThreshold defines how many 𝜎 a provider can be away from the mean
+	// without being considered faulty.
+	deviationThreshold = sdk.MustNewDecFromStr("2")
+)
+
+// PriceFeeder defines an interface an oracle with N exchange price
+// provider must implement.
+type PriceFeeder interface {
+	GetPrices(baseSymbols ...string) (map[string]sdk.Dec, error)
+	GetPrice(baseSymbol string) (sdk.Dec, error)
+	SubscribeSymbols(baseSymbols ...string) error
+}
```

Comment:
This code pattern is confusing, but I get we need interfaces for mocking. Let's [follow the codepattern in the price feeder here](https://github.com/umee-network/umee/blob/main/price-feeder/router/v1/oracle.go), where we add an `Oracle` interface as a new file under the directory we need to use it. In this case, that's under /relayer, e.g., 

```golang
package relayer

import (
	sdk "github.com/cosmos/cosmos-sdk/types"
)

// Oracle defines the Oracle interface contract that the v1 router depends on.
type Oracle interface {
	GetPrices(baseSymbols ...string) (map[string]sdk.Dec, error)
	GetPrice(baseSymbol string) (sdk.Dec, error)
	SubscribeSymbols(baseSymbols ...string) error
}
```

and then reference that interface instead 😄

Reply:
agreed, done at https://github.com/umee-network/peggo/pull/231/commits/34d218b1eb759a2ce3266afe36b2b790bb1a7c37

Can you check that again? @adamewozniak

Reply:
Much better and more readable 😄